runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -319,7 +319,8 @@ jvmtiGetFrameCount(jvmtiEnv* env,
 			if (NULL == targetThread) {
 				j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
 				j9object_t contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
-				vmFuncs->walkContinuationStackFrames(currentThread, contObject, &walkState);
+				J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, contObject);
+				vmFuncs->walkContinuationStackFrames(currentThread, continuation, &walkState);
 			} else
 #endif /* JAVA_SPEC_VERSION >= 19 */
 			{
```

Comment:
If the input `thread` is a carrier thread object, then invoking `walkStackFrames` on a carrier thread with a virtual thread mounted will go through the virtual thread's stack frames instead of the carrier thread's stack frames. Shouldn't we employ the below code-block to address earlier raised concern?
```
if (isVirtual(thread)) /* input thread is a virtual thread object */:
   if (null != targetThread->currentContinuation) /* mounted */: walkStackFrames
   else: /* !mounted */: walkContinuationStackFrames
else: /* input thread is a platform thread object */
   if (null != targetThread->currentContinuation) /* carrier thread with virtual thread mounted */: walkContinuationStackFrames
   else: walkStackFrames
```

Reply:
I'm doing that in another commit, not sure if everything should be in the same or separate commits/PRs

Reply:
To fix the core functional issue, all related changes should be included in the same PR.

Reply:
ok I'll mark it as WIP for now